### PR TITLE
Toolexecutionsuccessful rename & mark as required

### PR DESCRIPTION
**Pull request by harleenkohli**

### CODE REVIEW

File: src/Sarif/Core/Invocation.cs [hunk 1]
```diff
@@ -52,6 +52,8 @@ public partial class Invocation
                 invocation.ExecutableLocation.Uri = new Uri(assembly.Location);
             }
 
+            invocation.ExecutionSuccessful = true;
```
Comment (harleenkohli) [inline src/Sarif/Core/Invocation.cs]: >invocation.ExecutionSuccessful = true; [](start = 12, length = 38)

i am unsure if this is valid assignment! i couldnt figure out how to set this right, please cross check

File: src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs [hunk 2]
```diff
@@ -280,6 +280,7 @@ internal Fix CreateFix(FixVersionOne v1Fix)
 
                 invocation.ToolExecutionNotifications = toolNotifications;
                 invocation.ToolConfigurationNotifications = configurationNotifications;
+                invocation.ExecutionSuccessful = true;
```
Comment (harleenkohli) [inline src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs]: >invocation.ExecutionSuccessful = true; [](start = 16, length = 38)

our 1.0 schema doesnt seem to have an exitcode, so i set it to true

File: src/Sarif.Converters/FortifyFprConverter.cs [hunk 3]
```diff
@@ -110,6 +110,7 @@ public override void Convert(Stream input, IResultLogWriter output, OptionallyEm
 
             _invocation = new Invocation();
             _invocation.ToolExecutionNotifications = new List<Notification>();
+            _invocation.ExecutionSuccessful = true;
```
Comment (harleenkohli) [inline src/Sarif.Converters/FortifyFprConverter.cs]: >_invocation.ExecutionSuccessful = true; [](start = 12, length = 39)

again, perhaps there is a smarter way to figure out the right status here, i simply didnt have the right knowledge about this to improve. Please cross check if we can have some logic here.

File: src/Test.UnitTests.Sarif/TestData/PrereleaseCompatibilityTransformer/ExpectedOutputs/OneRunWithBasicInvocation.sarif [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+          "commandLine": "CodeScanner @collections.rsp",
+          "responseFiles": [
+            {
+              "uri": "collections.rsp",
+              "index": 0
+            }
+          ],
+          "startTimeUtc": "2016-07-16T14:18:25.000Z",
+          "endTimeUtc": "2016-07-16T14:19:01.000Z",
+          "executionSuccessful": false,
```
Comment (harleenkohli) [inline src/Test.UnitTests.Sarif/TestData/PrereleaseCompatibilityTransformer/ExpectedOutputs/OneRunWithBasicInvocation.sarif]: >"executionSuccessful": false, [](start = 10, length = 29)

i needed these new cases to ensure my new code paths are executed

File: src/Test.UnitTests.Sarif/TestData/PrereleaseCompatibilityTransformer/Inputs/OneRunWithBasicInvocation.sarif [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+          "commandLine": "CodeScanner @collections.rsp",
+          "responseFiles": [
+            {
+              "uri": "collections.rsp",
+              "index": 0
+            }
+          ],
+          "startTimeUtc": "2016-07-16T14:18:25.000Z",
+          "endTimeUtc": "2016-07-16T14:19:01.000Z",
+          "toolExecutionSuccessful": false,
```
Comment (harleenkohli) [inline src/Test.UnitTests.Sarif/TestData/PrereleaseCompatibilityTransformer/Inputs/OneRunWithBasicInvocation.sarif]: >"toolExecutionSuccessful": false, [](start = 10, length = 33)

i needed these new cases to ensure my new code paths are executed

Comment (harleenkohli) [reply src/Test.UnitTests.Sarif/TestData/PrereleaseCompatibilityTransformer/Inputs/OneRunWithBasicInvocation.sarif]: (also note schema is beta.2)

---
In reply to: [279148057](https://github.com/Microsoft/sarif-sdk/pull/1465#discussion_r279148057) [](ancestors = 279148057)

File: src/Test.UnitTests.Sarif/TestData/PrereleaseCompatibilityTransformer/Inputs/OneRunWithInvocationExitCode.sarif [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+          "commandLine": "CodeScanner @collections.rsp",
+          "responseFiles": [
+            {
+              "uri": "collections.rsp",
+              "index": 0
+            }
+          ],
+          "startTimeUtc": "2016-07-16T14:18:25.000Z",
+          "endTimeUtc": "2016-07-16T14:19:01.000Z",
+          "exitCode": 1,
```
Comment (harleenkohli) [inline src/Test.UnitTests.Sarif/TestData/PrereleaseCompatibilityTransformer/Inputs/OneRunWithInvocationExitCode.sarif]: > "exitCode": 1, [](start = 9, length = 15)

here we use exitcode to figure the right execution status

File: src/Sarif/Core/Invocation.cs (hunk 1 above)
Comment (michaelcfanning) [reply src/Sarif/Core/Invocation.cs]: this is fine for now. the thing actually driving tool execution needs to reset this to false if required.

---
In reply to: [279147950](https://github.com/Microsoft/sarif-sdk/pull/1465#discussion_r279147950) [](ancestors = 279147950)

File: src/Sarif/Writers/SarifLogger.cs [hunk 7]
```diff
@@ -555,6 +555,7 @@ public void LogToolNotification(Notification notification)
 
             _run.Invocations[0].ToolExecutionNotifications = _run.Invocations[0].ToolExecutionNotifications ?? new List<Notification>();
             _run.Invocations[0].ToolExecutionNotifications.Add(notification);
+            _run.Invocations[0].ExecutionSuccessful = true;
```
Comment (michaelcfanning) [inline src/Sarif/Writers/SarifLogger.cs]: >ExecutionSuccessful [](start = 32, length = 19)

You need to check the notification level here. If it is error then ExecutionSuccessful is set to false.

**Pull request merged**